### PR TITLE
Nested links

### LINKED ISSUE #4706: Nested links on Cards
Cards are themselves one big a tag but they also contain sub links that should also be a tags

See: https://css-tricks.com/nested-links/ for suggestions on how to achieve this

### PR DESCRIPTION
**Pull request by oliverlloyd**
## What does this change?
This PR fixes #4706 by adding support for nested links on cards.

We're using a similar approach to that used on Frontend where we have a faux link sitting outside the card which is absolutely positioned across the card surface. The nested sublinks are then brought to the top using z-index.

You can read more about this css pattern in [this great article on CSS Tricks](https://css-tricks.com/nested-links/)

## Why?
Having sublinks inside cards is an established pattern and one that we want to continue to support.

### CODE REVIEW

File: dotcom-rendering/src/web/components/Card/Card.tsx [hunk 1]
```diff
@@ -236,78 +236,54 @@ export const Card = ({
...
+								}
+								showSlash={
+									format.design === ArticleDesign.LiveBlog ||
+									showSlash
+								}
+								byline={byline}
+								showByline={showByline}
+							/>
+						</HeadlineWrapper>
+						<>
```
Comment (AshCorr) [inline dotcom-rendering/src/web/components/Card/Card.tsx:284]: Is this fragment necessary?

File: dotcom-rendering/src/web/components/Card/Card.tsx [hunk 2]
```diff
@@ -317,18 +293,40 @@ export const Card = ({
...
-					</ContentWrapper>
-				</CardLayout>
-				{/* If there are more than two sublinks break footer out of the headline column into a row below */}
-				{moreThanTwoSubLinks ? renderFooter({}) : <></>}
-			</CardWrapper>
-		</CardLink>
+						</>
+					</Flex>
+					<div>
+						{trailText && (
```
Comment (AshCorr) [inline dotcom-rendering/src/web/components/Card/Card.tsx:299]: Could be wrong but I think the trailtext should be hidden when there are nested links... I'm not sure if this is decided by the layout or if its the same for all cards regardless of layout.

**Review by AshCorr — APPROVED** [review]
Thank you!

File: dotcom-rendering/src/web/components/Card/Card.tsx (hunk 2 above)
Comment (oliverlloyd) [reply dotcom-rendering/src/web/components/Card/Card.tsx:299]: You do sometimes get both together
Screenshot 2022-05-03 at 09 50 15
I don't have a complete picture for the rules yet but they do seem to be layout specific. Indeed it looks like specific cards in certain positions have their own rules about what appears in them

Comment (AshCorr) [reply dotcom-rendering/src/web/components/Card/Card.tsx:299]: Ach, so complicated!

File: dotcom-rendering/src/web/components/Card/Card.tsx (hunk 1 above)
Comment (oliverlloyd) [reply dotcom-rendering/src/web/components/Card/Card.tsx]: Good spot. Fixing!

File: dotcom-rendering/src/web/components/Card/components/CardLink.tsx [hunk 3]
```diff
@@ -1,99 +1,23 @@
...
-	children,
-	linkTo,
-	format,
-	dataLinkName = 'article',
-}: Props) => (
-	<a href={linkTo} css={linkStyles(format)} data-link-name={dataLinkName}>
-		{children}
+export const CardLink = ({ linkTo, dataLinkName = 'article' }: Props) => (
+	<a href={linkTo} css={fauxLinkStyles} data-link-name={dataLinkName}>
+		{linkTo}
```
Comment (mxdvl) [inline dotcom-rendering/src/web/components/Card/components/CardLink.tsx:21]: Was it intentional to have the link content here?

File: dotcom-rendering/src/web/components/Card/components/CardLink.tsx [hunk 4]
```diff
@@ -1,99 +1,23 @@
...
-				}
-			`;
-	}
-};
+import { getZIndex } from '../../../lib/getZIndex';
+
+const fauxLinkStyles = css`
+	position: absolute;
+	${getZIndex('card-link')};
+	opacity: 0;
```
Comment (mxdvl) [inline dotcom-rendering/src/web/components/Card/components/CardLink.tsx:7]: Why make it transparent?

**Review by mxdvl — COMMENTED** [review]
Two small questions that will inform an upcoming PR.

File: dotcom-rendering/src/web/components/Card/components/CardLink.tsx (hunk 3 above)
Comment (oliverlloyd) [reply dotcom-rendering/src/web/components/Card/components/CardLink.tsx:21]: It was. I'm taking inspiration from Frontend here but I think a link isn't valid if it has no content. That said, I think this is for accessibility reasons and we still need to review our solution for accessibility.

Comment (mxdvl) [reply dotcom-rendering/src/web/components/Card/components/CardLink.tsx:21]: Lighthouse didn't care so we removed it in #xxx

**Pull request merged**